packages/l10n/README.md

```diff
@@ -97,3 +97,271 @@ First, download the data (`core.zip`) for a specific version from the [downloads
+This util is at the moment subject to change, hence marked as Draft. Use with caution in your Custom Application.
+
+#### Context
+
+As discussed under `applyTransformedLocalizedFields`, a `LocalizedString` is a value type reserved for a `Resource`, e.g a `Product`.
+
+```js
+const product = {
+  name: {
+    en: 'Milk',
+    de: 'Milsch,
+  }
+}
+```
+
+When rendering a Product in a view, we would like to render the value of `name` given the `language` of the UI.
+The `language` is a value controlled by the Merchant Center User (MC User) by updating the language switcher on the Application Bar of the Merchant Center.
+
+The language switcher emits a value that is intended to help us Application Developers display the Resource data (`Product` in our example) in the specified language. Given that the `product.name` has a value of the specified `language`, we can easily pick the value from the `product.name`.
+
+However there are scenarios where the MC User updates the language switcher to a value that our Product example has no matching value. What then? This is where `localize` plays its role.
+
+The `localize` util is authored with a couple of things in mind:
+
+1. Help us Application Developers remain resilient as we attempt to derive a value of `LocalizedString` given a specified language.
+2. Help us Application Developers remain consistent when rendering a value derived from `LocalizedString`
+
+Let's take a look at the examples below to put this to action.
+
+#### Example usage
+
+All examples below will use the following `Product`:
+
+```
+const product = {
+  name: {
+    en: 'Milk',
+    de: 'Milsch,
+  }
+}
+```
+
+##### Scenario 1
+
+Given that specified `language=de`
+
+```js
+const translatedName = localize({
+  obj: product,
+  key: 'name',
+  language: 'de',
+});
+
+console.log(translatedName);
+// 'Milsch'
+```
+
+##### Scenario 2
+
+Given that specified `language=sv`
+
+```js
+const translatedName = localize({
+  obj: product,
+  key: 'name',
+  language: 'sv',
+});
+
+console.log(translatedName);
+// 'Milk (EN)'
+```
+
+**What happened?**
+
+Our product has no value for the specified language `sv`, what `localize` will do for us as Application Developers is _pick the next available value_ from `product.name`.
+
+##### Scenario 3
+
+Given that specified `language=de-AT`
+
+```js
+const translatedName = localize({
+  obj: product,
+  key: 'name',
+  language: 'de-AT',
+});
+
+console.log(translatedName);
+// 'Milsch'
+```
+
+**What happened?**
+
+Our product has no value for the specified language `de-AT` so there is no match, but we still got `"Milsch"`.
+This is because `localize` attempts to determine a **primary language** on the input `language` and match that to the available values. Given that `de` is a [primary](https://en.wikipedia.org/wiki/IETF_language_tag) of `de-AT`, it determines that this is the _closest available value_ from `product.name`
```

Comment:
```suggestion
Our product has no matching value for the selected **data locale** `de-AT` but we still got `"Milch"`.
This is because `localize` attempts to determine a **primary language tag** on the option `locale` and match that to the available values. Given that `de` is a [primary](https://en.wikipedia.org/wiki/IETF_language_tag) of `de-AT`, it determines that this is the _closest available value_ from `product.name`
```